### PR TITLE
Extract kubelet network code into its own file

### PR DESCRIPTION
**Pull request by pmorie**
Continuing the effort to begin modularizing the kubelet, this PR extracts the networking code into its own file.

@kubernetes/sig-node cc @kubernetes/sig-network

### CODE REVIEW

File: pkg/kubelet/kubelet_network.go [hunk 1]
```diff
@@ -0,0 +1,305 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
Comment (euank) [inline pkg/kubelet/kubelet_network.go]: nit: 2016

**Comment by thockin** [comment]
nit: why do the file names stutter - they are already in a kubelet/ dir.

**Comment by pmorie** [comment]
@thockin they stutter because we wanted to show which are kubelet proper and which are support types.

**Comment by pmorie** [comment]
@thockin this is refactoring toward an interstitial state -- ultimately we don't expect to keep these files -- it's just about finding the natural decomposition of kubelet into modules so that we can factor out to true modules later.

File: pkg/kubelet/kubelet_test.go [hunk 2]
```diff
@@ -4145,88 +4034,7 @@ func TestDoesNotDeletePodDirsIfContainerIsRunning(t *testing.T) {
...
-		if err != nil {
-			t.Errorf("unexpected error: %v (%s)", test.name, err)
-		}
-		if !reflect.DeepEqual(shaper.ResetCIDRs, test.expectResetCIDRs) {
-			t.Errorf("[%s]\nexpected: %v, saw: %v", test.name, test.expectResetCIDRs, shaper.ResetCIDRs)
-		}
-	}
-}
-
+// TODO: why is this here and not in the bandwidth package?
```
Comment (yujuhong) [inline pkg/kubelet/kubelet_test.go]: +1 for moving `TestExtractBandwidthResources` to the bandwidth package

**Comment by yujuhong** [comment]
Looks good overall. Thanks for the cleanup!

**Comment by freehan** [comment]
LGTM. Thanks!

**Comment by pmorie** [comment]
:+1:

**Comment by pmorie** [comment]
I realized a deleted method snuck back in after rebase -- will fix.

**Pull request merged**